tests/test_api.py

```diff
@@ -315,3 +315,33 @@ async def content(request, page):
     assert response_one.text == "one"
     assert response_two.text == "two"
     assert respx.stats.call_count == 2
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "method_str, client_method_attr",
+    [
+        ("DELETE", "delete"),
+        ("delete", "delete"),
+        ("GET", "get"),
+        ("get", "get"),
+        ("HEAD", "head"),
+        ("head", "head"),
+        ("OPTIONS", "options"),
+        ("options", "options"),
+        ("PATCH", "patch"),
+        ("patch", "patch"),
+        ("POST", "post"),
+        ("post", "post"),
+        ("PUT", "put"),
+        ("put", "put"),
+    ],
+)
+async def test_can_add_by_lowercase_method(client, method_str, client_method_attr):
+    url = "https://example.org/"
+    content = {"spam": "lots", "ham": "no, only spam"}
+    async with MockTransport() as respx_mock:
+        request = respx_mock.add(method_str, url, content=content)
+        response = await getattr(client, client_method_attr)(url)
+        assert request.called is True
+        assert response.json() == content
```

Comment:
should we also be testing the `respx.add` way of doing this like in [test_mock.py::test_start_stop](https://github.com/lundberg/respx/blob/0ea0797f8f6001a9bb917deb126e60ff08ffe70d/tests/test_mock.py#L268)?

Reply:
I think the test is fine as is.